1-JavaScript/Week3/homework/ex4-shoppingCart.js

```diff
@@ -26,6 +26,13 @@ function addToShoppingCart(/* parameters go here */) {
 }
 
 // ! Test functions (plain vanilla JavaScript)
+function test0() {
+  console.log('Test 0: addShoppingCart() with intitial array of length 2 should not throw an index out of bounds exception');
```

Comment:
The assignment currently doesn't specify what should happen if the argument is undefined or `null`. The additional test implies that it should ignore the argument and just leave the shopping cart _as-is_. If I'm not mistaken, JavaScript will not throw an out-of-bounds exception anyway, it will just return `undefined` for an out-of-bounds array access.

```suggestion
  console.log('Test 0: addShoppingCart() called without an argument should leave the shopping cart unchanged');
```